frontend/src/Components/ConfirmInfo/ConfirmInfo.jsx

```diff
@@ -1,139 +1,174 @@
+import { useNavigate } from "react-router-dom";
+import { persistor } from "../../Redux/store";
+import { v4 as uuidv4 } from "uuid";
+
 function ConfirmInfo() {
-    const cartBooks = useSelector(addBookSelector);
-    const shippingInfo = useSelector(addConfirmSelector);
-    // const [phoneNumber,setPhoneNumber] = useState('')
-    // const [otp,setOtp] = useState('')
-    // Tính tổng số lượng và tổng giá tiền
-    const totalQuantity = cartBooks.reduce((acc, book) => acc + book.quantity, 0);
-    const subTotal = cartBooks.reduce((acc, book) => acc + book.quantity * book.newPrice, 0);
+  const cartBooks = useSelector(addBookSelector);
+  const shippingInfo = useSelector(addConfirmSelector);
+  const [user, setUser] = useState(null);
+  const [totalQuantity, setTotalQuantity] = useState(0);
+  const [subTotal, setSubTotal] = useState(0);
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
 
-    // const handleSendOTP = async ()=>{
-    //     const appVerify = window.recaptchaVerifier;
-    //     await firebase.auth().signInWithPhoneNumber(phoneNumber,appVerify)
-    //     .then((confirmationResult)=>{
-    //         window.confirmationResult = confirmationResult;
-    //         alert('OTP sended')
-    //     })
-    //     .catch((error)=>{
-    //         alert('Cant send OTP')
-    //     })
-    // }
-    // const handleVerifyOTP = ()=>{
-    //     window.confirmationResult.confirm(otp)
-    //     .then(()=>{
-    //         alert('OTP verified')
-    //     })
-    //     .catch(()=>{
-    //         alert('Cant verify OTP')
-    //     }
-    // )
+  // Tính tổng số lượng và tổng giá tiền
+  useEffect(() => {
+    if (cartBooks.length > 0) {
+      const totalQuantity = cartBooks.reduce(
+        (acc, book) => acc + book.quantity,
+        0
+      );
+      const subTotal = cartBooks.reduce(
+        (acc, book) => acc + book.quantity * book.newPrice,
+        0
+      );
+      setTotalQuantity(totalQuantity);
+      setSubTotal(subTotal);
+    } else {
+      setTotalQuantity(0);
+      setSubTotal(0);
+    }
+  }, [cartBooks]);
 
-    // }
-    // const setupCaptcha = ()=>{
-    //     window.recaptchaVerifier = new firebase.auth.RecaptchaVerifier('captcha-button',{
-    //         size: 'invisible',
-    //         defaultCountry: 'VN'
-    //     }) 
-    // }
-    // useEffect(()=>{
-    //     setupCaptcha()
-    // },[])
-    return (
-        <div className='confirmInfoContainer'>
-            <div className='confirmInfoHeader'>
-                <h1>Check out your cart</h1>
-            </div>
+  useEffect(() => {
+    const unsubscribe = auth.onAuthStateChanged((user) => {
+      if (user) {
+        setUser(user);
+      } else {
+        setUser(null);
+      }
+    });
+
+    return () => unsubscribe();
+  }, []);
+
+  const handleConfirmOrder = async () => {
+    if (!user) {
+      alert("Please log in to place an order.");
+      return;
+    }
+
+    try {
+      const orderId = uuidv4(); 
```

Comment:
All logic and declare payload  should be created a custom hook to component clearly